vortex-array/src/array/primitive/compute/fill_null.rs

```diff
@@ -0,0 +1,132 @@
+use vortex_buffer::Buffer;
+use vortex_dtype::{match_each_native_ptype, Nullability};
+use vortex_error::{vortex_err, VortexExpect, VortexResult};
+
+use crate::array::primitive::PrimitiveArray;
+use crate::array::{ConstantArray, PrimitiveEncoding};
+use crate::compute::FillNullFn;
+use crate::validity::{ArrayValidity, Validity};
+use crate::variants::PrimitiveArrayTrait;
+use crate::{ArrayDType, ArrayData, ArrayLen, IntoArrayData, ToArrayData};
+
+impl FillNullFn<PrimitiveArray> for PrimitiveEncoding {
+    fn fill_null(
+        &self,
+        array: &PrimitiveArray,
+        fill_value: vortex_scalar::Scalar,
+    ) -> VortexResult<ArrayData> {
+        let result_validity = match fill_value.dtype().nullability() {
+            Nullability::NonNullable => Validity::NonNullable,
+            Nullability::Nullable => Validity::AllValid,
+        };
+
+        if array.dtype().nullability() == Nullability::NonNullable
```

Comment:
I think it's enough to check array nullability - you don't have to do anything for non nullable array, whatever the fill value is